src/transformers/models/clip/modeling_clip.py

```diff
@@ -961,8 +1037,8 @@ def __init__(self, config: CLIPConfig):
         self.text_embed_dim = text_config.hidden_size
         self.vision_embed_dim = vision_config.hidden_size
 
-        self.text_model = CLIPTextTransformer(text_config)
-        self.vision_model = CLIPVisionTransformer(vision_config)
+        self.text_model = CLIPTextTransformer(text_config, attn_implementation=config._attn_implementation)
+        self.vision_model = CLIPVisionTransformer(vision_config, attn_implementation=config._attn_implementation)
```

Comment:
I definitely think there's some kind of a misunderstanding here. `CLIPTextTransformer` inherits only from `nn.Module` and not from `PretrainedModel`, so I don't know how it would have access to methods like `from_config`. But if I have missing something, please correct me.

Reply:
Ah, my bad, I was confusing it with the CLIPTextModel class. 

So the thing that makes this a bit tricky, is that in the rest of the repo we assume that we control the attention implementation from the config. The way it's done when we have other modular models like this is: 
* The parent config has its `_attn_implementation` attribute set, either when the config is created, or when the parent model is created e.g. `CLIPModel.from_pretrained(checkpoint, attn_implementation="sdpa")`
* The children models' configs have this parameter set [in `from_config` calls](https://github.com/huggingface/transformers/blob/f4dc26d46687f5f4baf3fe64a1d87cafefbeec53/src/transformers/models/llava/modeling_llava.py#L244). This means that in the e.g. language_model, `config._attn_implementation` is the same as the parent class e.g. SDPA.  

At the moment, this means that the text config and model config could have different attn_implementation values set. This becomes a problem when someone wants to load `CLIPTextModel` on its own with a specific attention implementation. For example, this won't have the expected behaviour:

```py
from transformers import CLIPTextModel, CLIPTextConfig

config = CLIPTextConfig(attn_implementation="sdpa")
model = CLIPTextModel(config)
```

I see two possible ways around this either: 

* `attn_implementation` in `CLIPTextTransformer` and `CLIPVisionTransformer` defaults to `None`. If it's unset, it takes the value from the config. 
* Instead of using `CLIPTextTransformer` and `CLIPVisionTransformer` here, we use `CLIPTextModel.from_config(config.text_config, attn_implementation=config._attn_implementation)`

The first is easier, the second is more likely to be forwards compatible cc @ArthurZucker WDYT?

Reply:
@ArthurZucker a gentle ping for your feedback on this :-)

Reply:
Sorry for the delay, having a llook

Reply:
@ArthurZucker did you mean to submit feedback? Just checking.

Reply:
Done.